WordPressKitTests/Utilities/URLSessionHelperTests.swift

```diff
@@ -0,0 +1,130 @@
+import Foundation
+import XCTest
+import OHHTTPStubs
+
+@testable import WordPressKit
+
+class URLSessionHelperTests: XCTestCase {
+
+    override func tearDown() {
+        super.tearDown()
+        HTTPStubs.removeAllStubs()
+    }
+
+    func testConnectionError() async throws {
+        stub(condition: isPath("/hello")) { _ in
+            HTTPStubsResponse(error: URLError(.serverCertificateUntrusted))
+        }
+
+        let result = await URLSession.shared.perform(request: .init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
+        do {
+            _ = try result.get()
+            XCTFail("The above call should throw")
+        } catch let WordPressAPIError<TestError>.connection(error) {
+            XCTAssertEqual(error.code, URLError.Code.serverCertificateUntrusted)
+        } catch {
+            XCTFail("Unknown error: \(error)")
+        }
+    }
+
+    func test200() async throws {
+        stub(condition: isPath("/hello")) { _ in
+            HTTPStubsResponse(data: "success".data(using: .utf8)!, statusCode: 200, headers: nil)
+        }
+
+        let result = await URLSession.shared.perform(request: .init(url: URL(string: "https://wordpress.org/hello")!), errorType: TestError.self)
+
+        // The result is a successful result. This line should not throw
+        _ = try result.get()
+
+        let expectation = expectation(description: "API call returns a successful result")
+        _ = result
+            .assessStatusCode { result in
+                XCTAssertEqual(String(data: result.body, encoding: .utf8), "success")
+                expectation.fulfill()
+                return result
+            } failure: { _ in
+                // Do nothing
+                return nil
+            }
+        await fulfillment(of: [expectation])
```

Comment:
Have you considered making the code synchronous?

```suggestion
        _ = result
            .assessStatusCode { result in
                XCTAssertEqual(String(data: result.body, encoding: .utf8), "success")
                return result
            } failure: { _ in
                XCTFail("Expected result to be successful and this branch not to run")
                return nil
            }
```

Reply:
If neither closures of `assessStatusCode` is called, the updated test would pass. But the original test case would fail and catch the bug.